apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/blockchain/TransactionService.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c)  2018-2020. Apollo Foundation.
+ */
+
+package com.apollocurrency.aplwallet.apl.core.service.blockchain;
+
+import com.apollocurrency.aplwallet.api.v2.model.TxReceipt;
+import com.apollocurrency.aplwallet.apl.core.entity.blockchain.Transaction;
+import com.apollocurrency.aplwallet.apl.core.model.TransactionDbInfo;
+
+import java.util.List;
+
+/**
+ * @author andrew.zinchenko@gmail.com
+ */
+public interface TransactionService {
+    Transaction findTransaction(long transactionId);
+
+    Transaction findTransaction(long transactionId, int height);//cross sharding
```

Comment:
Better to change name of the method to note that search affects shard datasources

Reply:
added CrossSahrding suffix to method name;